src/shared/components/query/QueryStore.spec.ts

```diff
@@ -122,4 +124,92 @@ describe("QueryStore", ()=>{
             // },1000)  
         });
     });
-});
\ No newline at end of file
+});
+describe("QueryStore", ()=>{
+    describe("Volcano Plot related functions", ()=>{
+        const genesets: Geneset[] = [{
+        	description: "AKT_UP.V1_DN",
+        	genesetId: "AKT_UP.V1_DN",
+        	name: "AKT_UP.V1_DN",
+        	refLink: "http://www.example.com",
+        	representativePvalue: 0.0736,
+        	representativeScore: 0.1986
+        },
+        {
+        	description: "AKT_UP.V1_UP",
+        	genesetId: "AKT_UP.V1_UP",
+        	name: "AKT_UP.V1_UP",
+        	refLink: "http://www.example.com",
+        	representativePvalue: 0.0862,
+        	representativeScore: 0.3945
+        }];
+        const hierarchyData: any[] = [{
+        	nodeId: 2, 
+        	nodeName: "C6 Oncogenic Signatures",
+        	parentId: 1,
+        	parentNodeName: "Molecular Signatures Database v6.0",
+        	genesets: genesets
+        	}];
+		const hierarchyDataNoGenesets: any[] = [{
+			nodeId: 2, 
+			nodeName: "C6 Oncogenic Signatures",
+			parentId: 1,
+			parentNodeName: "Molecular Signatures Database v6.0",
+		}];
+        let volcanoPlotTableData = buildTableData(hierarchyData);
+        let volcanoPlotTableDataNoGenesets = buildTableData(hierarchyDataNoGenesets);
+        let maxYValue = generateMaxYValue(volcanoPlotTableData);
+        let maxYValueUndefined = generateMaxYValue(undefined);
+        let map_genesets_selected_volcano = new ObservableMap<boolean>();
+        map_genesets_selected_volcano.set("AKT_UP.V1_DN", true);
+        let graphData = generateVolcanoPlotData(volcanoPlotTableData, map_genesets_selected_volcano);
+        let graphDataUndefined = generateVolcanoPlotData(undefined, map_genesets_selected_volcano);
+        it("builds the data for the volcano plot table with the correct values", ()=>{
+            assert.deepEqual(
+                volcanoPlotTableData,
+                [{
+                	description: "AKT_UP.V1_DN",
+                	genesetId: "AKT_UP.V1_DN",
+                	name: "AKT_UP.V1_DN",
+                	refLink: "http://www.example.com",
+                	representativePvalue: 0.0736,
+                	representativeScore: 0.1986
+                },
+                {
+                	description: "AKT_UP.V1_UP",
+                	genesetId: "AKT_UP.V1_UP",
+                	name: "AKT_UP.V1_UP",
+                	refLink: "http://www.example.com",
+                	representativePvalue: 0.0862,
+                	representativeScore: 0.3945
+                }]
+            );
+        });
+        it("calculates the correct max Y value for the volcano plot", ()=>{
+            assert.equal(maxYValue, -(Math.log(0.0736)/Math.log(10)));
```

Comment:
you want to try to put the call to the code you are testing (generateMaxYValue) inside of the call to it.  this makes it a lot clearer.